gslib/commands/cp.py

```diff
@@ -282,82 +276,79 @@
     Copying file://obj [Content-Type=text/plain]...
     Uploading   gs://your-bucket/obj:                                182 b/182 B
 
-    If the checksum didn't match the service would instead reject the upload and
-    gsutil would print a message like:
+    If the checksums don't match, the service rejects the upload and
+    gsutil prints the following message:
 
     BadRequestException: 400 Provided MD5 hash "VgyllJgiiaRAbyUUIqDMmw=="
     doesn't match calculated MD5 hash "7gyllJgiiaRAbyUUIqDMmw==".
 
-  Even if you don't do this gsutil will delete the object if the computed
-  checksum mismatches, but specifying the Content-MD5 header has several
-  advantages:
+  Specifying the Content-MD5 header has several advantages:
 
-  1. It prevents the corrupted object from becoming visible at all, whereas
-     otherwise it would be visible for 1-3 seconds before gsutil deletes it.
+  1. It prevents the corrupted object from becoming visible. Otherwise, the
+     object is visible for 1-3 seconds before gsutil deletes it.
 
   2. If an object already exists with the given name, specifying the
-     Content-MD5 header will cause the existing object never to be replaced,
-     whereas otherwise it would be replaced by the corrupted object and then
+     Content-MD5 header prevents the existing object from being replaced.
+     Otherwise, the existing object is replaced by the corrupted object and
      deleted a few seconds later.
 
-  3. It will definitively prevent the corrupted object from being left in
-     the cloud, whereas the gsutil approach of deleting after the upload
-     completes could fail if (for example) the gsutil process gets ^C'd
-     between upload and deletion request.
+  3. If the gsutil process is interrupted or fails before it can delete the
+     corrupted object after uploading it, specifying the Content-MD5 header
```

Comment:
Suggestion: If you don't specify the Content-MD5 header, it's possible for the gsutil process to complete the upload but then be interrupted or fail before it can delete the corrupted object, leaving the corrupted object in the cloud.

Reply:
Done.